src/CoreWCF.Primitives/src/CoreWCF/Security/FederatedSecurityTokenManager.cs

```diff
@@ -0,0 +1,589 @@
+        /// <param name="outOfBandTokenResolver">Token resolver that resolves any out-of-band tokens.</param>
+        /// <returns>Instance of Security Token Authenticator.</returns>
+        /// <exception cref="ArgumentNullException">'tokenRequirement' parameter is null.</exception>
+        /// <exception cref="NotSupportedException">No Authenticator is registered for the given token type.</exception>
+        public override SecurityTokenAuthenticator CreateSecurityTokenAuthenticator(SecurityTokenRequirement tokenRequirement, out SecurityTokenResolver outOfBandTokenResolver)
+        {
+            if (tokenRequirement == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(tokenRequirement));
+            }
+
+            outOfBandTokenResolver = null;
+            string tokenType = tokenRequirement.TokenType;
+            //
+            // When the TokenRequirement.TokenType is null, we treat this as a SAML issued token case. It may be SAML 1.1 or SAML 2.0.
+            //
+            if (string.IsNullOrEmpty(tokenType))
+            {
+                return CreateSamlSecurityTokenAuthenticator(tokenRequirement, out outOfBandTokenResolver);
+            }
+
+            //
+            // When the TokenType is set, build a token authenticator for the specified token type.
+            //
+            SecurityTokenHandler securityTokenHandler = SecurityTokenHandlers[tokenType];
+
+            // Check for a registered authenticator
+            SecurityTokenAuthenticator securityTokenAuthenticator;
+            if ((securityTokenHandler != null) && (securityTokenHandler.CanValidateToken))
+            {
+                outOfBandTokenResolver = GetDefaultOutOfBandTokenResolver();
+
+                if (StringComparer.Ordinal.Equals(tokenType, SecurityTokenTypes.UserName))
+                {
+                    if (!(securityTokenHandler is UserNameSecurityTokenHandler upSecurityTokenHandler))
+                    {
+                        throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(
+                            new InvalidOperationException(SR.Format(SR.ID4072, securityTokenHandler.GetType(), tokenType, typeof(UserNameSecurityTokenHandler))));
+                    }
+                    securityTokenAuthenticator = new WrappedUserNameSecurityTokenAuthenticator(upSecurityTokenHandler, _exceptionMapper);
+                }
+                else if (StringComparer.Ordinal.Equals(tokenType, SecurityTokenTypes.Kerberos))
+                {
+                    securityTokenAuthenticator = CreateInnerSecurityTokenAuthenticator(tokenRequirement, out outOfBandTokenResolver);
+                }
+                //TODO: not sure if this is supported
+                //else if (StringComparer.Ordinal.Equals(tokenType, SecurityTokenTypes.Rsa))
+                //{
```

Comment:
Leave the else if clause in there but have the implementation block throw PlatformNotSupportedException.